extensions/resource-deployment/src/ui/eulaWizard/eulaWizard.ts

```diff
@@ -0,0 +1,153 @@
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import { INotebookService, Notebook } from '../../services/notebookService';
+import { IToolsService } from '../../services/toolsService';
+import { Model } from '../model';
+import { InputComponents, setModelValues } from '../modelViewUtils';
+import { WizardBase } from '../wizardBase';
+import { DeploymentProvider, DeploymentType, EulaWizardDeploymentProvider, instanceOfEulaWizardDeploymentProvider, NotebookWizardInfo, ResourceType } from '../../interfaces';
+import { IPlatformService } from '../../services/platformService';
+import { NotebookWizardAutoSummaryPage } from './eulaWizardAutoSummaryPage';
+import { EulaWizardPage } from './eulaWizardPage';
+import { EulaInformation, EulaSettingsPage } from './eulaSettingsPage';
+//import { EulaSettingsPage } from './eulaSettingsPage';
+
+const localize = nls.loadMessageBundle();
+
+export class EulaWizard extends WizardBase<EulaWizard, EulaWizardPage, Model> {
+
+	private _inputComponents: InputComponents = {};
+
+	public get notebookService(): INotebookService {
+		return this._notebookService;
+	}
+
+	public get platformService(): IPlatformService {
+		return this._platformService;
+	}
+
+	public get wizardInfo(): NotebookWizardInfo {
+		return this._provider.eulaWizard;
+	}
+
+	public get provider(): EulaWizardDeploymentProvider {
+		return this._provider;
+	}
+
+	public get inputComponents(): InputComponents {
+		return this._inputComponents;
+	}
+
+	constructor(private _resourceType: ResourceType | undefined, private _provider: EulaWizardDeploymentProvider, private _notebookService: INotebookService, private _platformService: IPlatformService, toolsService: IToolsService, private _settingsPreset?: EulaInformation) {
+		super(_provider.eulaWizard.title, '', new Model(), toolsService);
+		if (this._provider.eulaWizard.codeCellInsertionPosition === undefined) {
+			this._provider.eulaWizard.codeCellInsertionPosition = 0;
+		}
+		this.wizardObject.doneButton.label = _provider.eulaWizard.actionText || this.wizardObject.doneButton.label;
+	}
+
+	public get deploymentType(): DeploymentType | undefined {
+		return this._provider.eulaWizard.type;
+	}
+
+	public get resourceType(): ResourceType {
+		return this._resourceType!;
+	}
+
+	protected initialize(): void {
+		this.setPages(this.getPages());
+		this.wizardObject.generateScriptButton.hidden = true;
+		this.wizardInfo.actionText = this.wizardInfo.actionText || localize('eulaWizard.ScriptToNotebook', "Script to Notebook");
+		this.wizardObject.doneButton.label = this.wizardInfo.actionText;
+	}
+
+	protected onCancel(): void {
+	}
+
+	protected async onOk(): Promise<void> {
+		await setModelValues(this.inputComponents, this.model);
```

Comment:
Can we move this code from NotebookPageWizard to a untils class with appropriate refactoring and use it there as well as here?